web-admin/src/routes/[organization]/[project]/[dashboard]/+page.svelte

```diff
@@ -42,21 +47,60 @@
   }
 
   async function getDashboardsAndInvalidate() {
-    return invalidateProjectQueries(
-      queryClient,
-      await getDashboardsForProject($project.data)
-    );
+    const dashboardListings = await getDashboardsForProject($project.data);
+    const dashboardNames = dashboardListings.map((listing) => listing.name);
+    return invalidateProjectQueries(queryClient, dashboardNames);
   }
+
+  // Here we check to see if a dashboard is valid by looking at `DashboardListItem.isValid`
+  // As is the case in `Breadcrumbs.svelte`, there are two queries we compose to get the dashboard list items,
+  // and we should hide this complexity in a custom hook.
+  // We avoid calling `GetCatalogEntry` to check for dashboard validity because that would trigger a 404 page.
+  $: dashboardFiles = createRuntimeServiceListFiles(
+    $runtime?.instanceId,
+    {
+      glob: "dashboards/*.yaml",
+    },
+    {
+      query: {
+        placeholderData: undefined,
+        enabled: !!project && !!$runtime?.instanceId,
+      },
+    }
+  );
+  $: dashboardCatalogEntries = createRuntimeServiceListCatalogEntries(
+    $runtime?.instanceId,
+    {
+      type: "OBJECT_TYPE_METRICS_VIEW",
+    },
+    {
+      query: {
+        placeholderData: undefined,
+        enabled: !!project && !!$runtime?.instanceId,
+      },
+    }
+  );
+  $: dashboardListItems = getDashboardListItemsFromFilesAndCatalogEntries(
```

Comment:
Why do we need the full list here? Should be enough to just fetch a single file and single catalog entry and get the validity status from that.

Reply:
This can be fixed after the release. It is just optimization.